pkg/core/xds/rules.go

```diff
@@ -156,33 +166,49 @@ func BuildRules(list []PolicyItem) Rules {
 		return rules[i].Subset.NumPositive() > rules[j].Subset.NumPositive()
 	})
 
-	return rules
+	return rules, nil
+}
+
+func Merge(dst, src proto.Message) error {
+	dstBytes, err := json.Marshal(dst)
+	if err != nil {
+		return err
+	}
+	srcBytes, err := json.Marshal(src)
+	if err != nil {
+		return err
+	}
+	result, err := jsonpatch.MergeMergePatches(dstBytes, srcBytes)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(result, dst)
```

Comment:
This doesn't look very efficient no?

Reply:
Yes, it doesn't. I can change `Merge` function to accept a list of confs (not just a pair) and merge them all at once without extra marshal/unmarshal operations